python/cuml/ensemble/randomforestregressor.pyx

```diff
@@ -625,22 +620,25 @@ class RandomForestRegressor(Base):
             ndarray, cuda array interface compliant array like CuPy
         y: NumPy
            Dense vector (int) of shape (n_samples, 1)
+        algo : string name of the algo from (from algo_t enum)
+               This is optional and required only while performing the
+               predict operation on the GPU.
+               'NAIVE' - simple inference using shared memory
+               'TREE_REORG' - similar to naive but trees rearranged to be more
+                              coalescing-friendly
+               'BATCH_TREE_REORG' - similar to TREE_REORG but predicting
+                                    multiple rows per thread block
         Returns
         ----------
         mean_square_error : float or
         median_abs_error : float or
         mean_abs_error : float
         """
         cdef uintptr_t X_ptr, y_ptr
-        X_m, X_ptr, n_rows, n_cols, _ = \
-            input_to_dev_array(X, order='C')
-        y_m, y_ptr, _, _, _ = input_to_dev_array(y)
+        y_m, y_ptr, n_rows, _, _ = input_to_dev_array(y)
```

Comment:
Need to check dtype or dimensions?

Reply:
the dimension gets checked inside the predict function, i will add a check for dtype in it as well.